lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runime/azbatch/driver/AzureBatchResourceManager.java

```diff
@@ -0,0 +1,210 @@
+import org.apache.reef.tang.annotations.Parameter;
+import org.apache.reef.tang.exceptions.BindException;
+import org.apache.reef.tang.formats.ConfigurationSerializer;
+import org.apache.reef.util.CollectionUtils;
+import org.apache.reef.wake.EventHandler;
+import org.apache.reef.wake.remote.address.LocalAddressProvider;
+import org.joda.time.DateTime;
+
+import javax.inject.Inject;
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * A resource manager that uses threads to execute containers.
+ */
+@Private
+@DriverSide
+public final class AzureBatchResourceManager {
+
+  private static final Logger LOG = Logger.getLogger(AzureBatchResourceManager.class.getName());
+  private final EventHandler<ResourceAllocationEvent> resourceAllocationHandler;
+  private final EventHandler<NodeDescriptorEvent> nodeDescriptorHandler;
+  private String localAddress;
+  private final REEFFileNames fileNames;
+  private final ConfigurationSerializer configurationSerializer;
+  private final double jvmHeapFactor;
+  private final String azureBatchAccountUri;
+  private final String azureBatchAccountName;
+  private final String azureBatchAccountKey;
+  private final String azureBatchPoolId;
+  private final String applicationId;
+  private final AzureUploader azureUploader;
+
+  @Inject
+  AzureBatchResourceManager(
+      @Parameter(RuntimeParameters.ResourceAllocationHandler.class) final EventHandler<ResourceAllocationEvent> resourceAllocationHandler,
+      @Parameter(RuntimeParameters.NodeDescriptorHandler.class) final EventHandler<NodeDescriptorEvent> nodeDescriptorHandler,
+      final LocalAddressProvider localAddressProvider,
+      final REEFFileNames fileNames,
+      final ConfigurationSerializer configurationSerializer,
+      @Parameter(JVMHeapSlack.class) final double jvmHeapSlack,
+      final AzureUploader azureUploader,
+      @Parameter(AzureBatchAccountUri.class) final String azureBatchAccountUri,
+      @Parameter(AzureBatchAccountName.class) final String azureBatchAccountName,
+      @Parameter(AzureBatchAccountKey.class) final String azureBatchAccountKey,
+      @Parameter(AzureBatchPoolId.class) final String azureBatchPoolId) {
+    this.resourceAllocationHandler = resourceAllocationHandler;
+    this.nodeDescriptorHandler = nodeDescriptorHandler;
+    this.localAddress = localAddressProvider.getLocalAddress();
+    this.fileNames = fileNames;
+    this.configurationSerializer = configurationSerializer;
+    this.jvmHeapFactor = 1.0 - jvmHeapSlack;
+    this.azureUploader = azureUploader;
+    this.azureBatchAccountKey = azureBatchAccountKey;
+    this.azureBatchAccountName = azureBatchAccountName;
+    this.azureBatchAccountUri = azureBatchAccountUri;
+    this.azureBatchPoolId = azureBatchPoolId;
+    this.applicationId = "EvaluatorJob-"
```

Comment:
Actually, you can find the ID of the job by looking at "AZ_BATCH_JOB_ID" environment variable. Or, you can pass it via Driver config. Either approach is fine with me.
https://docs.microsoft.com/en-us/azure/batch/batch-compute-node-environment-variables

Reply:
mentioned in Note. This will be implemented in follow up task.